### PR TITLE
Handle incremental tables being empty

### PR DESCRIPTION
**Pull request by tommyh**
If the incremental table is empty, ie: `dim_dbt__models`, then `max(artifact-generated_at)` will be `NULL`, which means the `dbt_models_incremental` CTE will return 0 rows.

This fix is based on the work that @timothyjang123 did in the following [issue](https://github.com/brooklyn-data/dbt_artifacts/issues/64).

Notes:

1) We weren't sure how to run the `integration_tests` for the `dbt_artifacts` package in isolation, but we are using this fork in our dbt project and it is working successfully.  If there is something we should do to verify our change doesn't break anything with the package in general, we'd be happy todo that.

2) This fix works, but the first time you run it `dbt test` might fail.  The issue in more detail: In our project, we've run `dbt run` several times, so the base `artifacts` table has several runs in it.  If the `max(artifact-generated_at)` is `NULL` and the query falls back to the jan-1-1970 date, then the `dbt_models_incremental` will return rows for several runs.  This means the `unique_fct_dbt__critical_path_node_id`, `unique_fct_dbt__latest_full_model_executions_node_id`, and `unique_fct_dbt__model_executions_model_execution_id` will fail because the count on index will be greater then 1.  I'm not sure if this means that there is a better fix possible.  But I thought it'd be good to open up the PR as-is, as a jumping off point.

### CODE REVIEW

File: models/fct_dbt__exposures_updates.sql [hunk 1]
```diff
@@ -33,7 +33,7 @@ exposures_latest as (
         package_name,
         output_feeds
     from exposures_record
-    where artifact_generated_at = (select max(artifact_generated_at) from exposures_record)
+    where artifact_generated_at = (select ifnull(max(artifact_generated_at), '1970-01-01 00:00:00 +0000') from exposures_record)
```
Comment (NiallRees) [inline models/fct_dbt__exposures_updates.sql]: I'm not sure this is correct - this isn't incremental logic, it's just to pull the latest exposures :)

**Review by NiallRees — CHANGES_REQUESTED** [review]
Thank you @tommyh! I think this is good to go with the exception of `fct_dbt__exposures_updates.sql`, see my comment.

File: models/fct_dbt__exposures_updates.sql (hunk 1 above)
Comment (tommyh) [reply models/fct_dbt__exposures_updates.sql]: good call - should have looked a bit deeper at the find/replace results.  I'll remove that change.

**Comment by tommyh** [comment]
I reverted the change for `fct_dbt__exposures_updates.sql`.

Question: did you see my second note about "dbt test" failing for the first time using this PR, because it will pickup multiple artifacts until the `max(artifact_generated_at)` is actually set.  Should I be doing something better here?

File: models/incremental/dim_dbt__exposures.sql [hunk 2]
```diff
@@ -18,7 +18,7 @@ dbt_models_incremental as (
 
     {% if is_incremental() %}
     -- this filter will only be applied on an incremental run
-    where artifact_generated_at > (select max(artifact_generated_at) from {{ this }})
+    where artifact_generated_at > (select ifnull(max(artifact_generated_at), '1970-01-01 00:00:00 +0000') from {{ this }})
```
Comment (NiallRees) [inline models/incremental/dim_dbt__exposures.sql:21]:
```suggestion
    where coalesce(artifact_generated_at > (select max(artifact_generated_at) from {{ this }}), true)
```

What do you think?

**Comment by NiallRees** [comment]
> 2\. unique_fct_dbt__critical_path_node_id

I don't think I follow on the note about test failures. Please could you outline a series of steps to reproduce those errors?

Also, I've now added some integration test configuration to the repo, so if you're able to merge the main branch into this we can run the tests 🎉

**Comment by NiallRees** [comment]
Hey @tommyh just checking you saw my above message. Let me know if I can help get this over the line.

**Comment by NiallRees** [comment]
I'm going to close this out as I think it's solved by running a full refresh.

**Comment by ghost** [comment]
@NiallRees The issue with incremental loading into empty tables does still seem to exist and does seem to (mostly) be fixed with the addition of the proposed changes above. When I did a little testing, the ifnull/coalesce did solve it for some of the incremental table loads. It would be great to include these changes so it's a little more friendly to users starting out from scratch with the package.

**Comment by NiallRees** [comment]
Thanks for the insight @brycebaker-centura. Does using a `--full-refresh` also solve this? If so, I'm inclined to just update the documentation rather than add this logic which feels like a little brittle to me.

**Comment by ghost** [comment]
@NiallRees It does seem to populate some of the tables, but not all of them that I would expect so I'm hesitant to say --full-refresh alone is a complete answer (see comment here: https://github.com/brooklyn-data/dbt_artifacts/issues/102#issuecomment-1064489492). Could be that I'm misunderstanding what should be populated and when, which we could also address in documentation. Happy to contribute a README rewrite since I've first hand experience trying to wrap my head around using the package for the first time. Feel free to ping me in the dbt Slack also if needed.

**Pull request closed without merging**